sonic_data_client/db_client.go

```diff
@@ -803,12 +813,16 @@ func TableData2Msi(tblPath *tablePath, useKey bool, op *string, msi *map[string]
 	return nil
 }
 
-func msi2TypedValue(msi map[string]interface{}) (*gnmipb.TypedValue, error) {
+func Msi2TypedValue(msi map[string]interface{}) (*gnmipb.TypedValue, error) {
+	log.V(4).Infof("State of map after adding redis data %v", msi)
 	jv, err := emitJSON(&msi)
 	if err != nil {
 		log.V(2).Infof("emitJSON err %s for  %v", err, msi)
 		return nil, fmt.Errorf("emitJSON err %s for  %v", err, msi)
 	}
+	if jv == nil { // json is nil because of potential panic happen
+		return nil, fmt.Errorf("emitJSON failed due to panic")
```

Comment:
>panic [](http://example.com/codeflow?start=49&length=5)

panic may not be the only reason, could the error message be more broader or general like "emitJSON failed due to panic, err=..."

Reply:
The idea here is that on 823, both the json value and the err value would be nil, since if emitJSON has panicked and we have recovered, the values of jv and err would be the zero values which is nil. I can instead say "emitJSON was unable to grab json value of map due to potential panic". In other cases where emitJSON would have failed to get json value we would see that err != nil.